### PR TITLE
User Feedback Model and Endpoints

### LINKED ISSUE #204: Create staff feedback model
## Description

Create the staff feedback model and any necessary endpoints. Name this model _UserFeedback_. Here is some context that will help with writing endpoints and making any other additions to existing models (i.e. user model):
- We will need to display all of the feedback that a user has received on their profile page
- We will need to display some average number of stars (not sure about average vs. median or how we will weigh multiple ratings given by different staff or admin for the contributor's work on the same story) on each user's profile page and potentially in the directory
- We might display all of the feedback that a user has received for a pitch on their profile page (filtering via dropdown)

## Schema
- Staff or admin giving the feedback: User._id
- User receiving the feedback: User._id
- Associated pitch: Pitch._id
- Number of stars (min: 1, max: 5): Int
- Reasoning: String

### PR DESCRIPTION
**Pull request by zorazrr**
## Summary

Creates the user feedback model and endpoints.

Closes #204 

## Changes

- Add userFeedback model
- Add userFeedback endpoints

## Requests / Responses

**Request**
POST `/api/userFeedback` Returns the user feedback created

```
{
    "staffId": "6181f2b2bccaa891952ce94b",
    "userId": "6181f2bdfe8a2e3f44f13c18",
    "pitchId": "6181f343c6cf988c1a05b941",
    "stars": 5,
    "reasnoning": "pog"
}
```

**Response**
HTTP/1.1 200 OK

```
{
    "message": "Successfully created new userFeedback",
    "success": true,
    "result": {
        "stars": 5,
        "reasnoning": "pog",
        "_id": "619ae7f8a78eee20210fc94e",
        "staffId": "6181f2b2bccaa891952ce94b",
        "userId": "6181f2bdfe8a2e3f44f13c18",
        "pitchId": "6181f343c6cf988c1a05b941",
        "createdAt": "2021-11-22T00:44:40.329Z",
        "updatedAt": "2021-11-22T00:44:40.329Z",
        "__v": 0
    }
}
```

**Request**
GET `/api/userFeedback` Returns all user feedback

**Response**
HTTP/1.1 200 OK

```
{
    "message": "Successfully retrieved all user feedback.",
    "success": true,
    "result": [
        {
            "stars": 5,
            "reasnoning": "pog",
            "_id": "619ae7f8a78eee20210fc94e",
            "staffId": "6181f2b2bccaa891952ce94b",
            "userId": "6181f2bdfe8a2e3f44f13c18",
            "pitchId": "6181f343c6cf988c1a05b941",
            "createdAt": "2021-11-22T00:44:40.329Z",
            "updatedAt": "2021-11-22T00:44:40.329Z",
            "__v": 0
        },
        {
            "stars": 3,
            "reasnoning": "Lookin good!",
            "_id": "619aeb04a78eee20210fc94f",
            "staffId": "61847e07085c86081c505455",
            "userId": "618320d08e04de0cd0d14e43",
            "pitchId": "61833ff67aad832eb4672222",
            "createdAt": "2021-11-22T00:57:40.864Z",
            "updatedAt": "2021-11-22T00:57:40.864Z",
            "__v": 0
        },
        {
            "stars": 5,
            "reasnoning": "reasoning is uwu",
            "_id": "619af458d10c1121e4926dac",
            "staffId": "6181f2b2bccaa891952ce94b",
            "userId": "6181f2bdfe8a2e3f44f13c18",
            "pitchId": "6181f343c6cf988c1a05b941",
            "createdAt": "2021-11-22T01:37:28.342Z",
            "updatedAt": "2021-11-22T01:37:28.342Z",
            "__v": 0
        }
    ]
}
```

**Request**
GET `/api/userFeedback/:id` Returns a single user feedback

**Response**
HTTP/1.1 200 OK

```
{
    "success": true,
    "result": {
        "stars": 5,
        "reasnoning": "pog",
        "_id": "619ae7f8a78eee20210fc94e",
        "staffId": "6181f2b2bccaa891952ce94b",
        "userId": "6181f2bdfe8a2e3f44f13c18",
        "pitchId": "6181f343c6cf988c1a05b941",
        "createdAt": "2021-11-22T00:44:40.329Z",
        "updatedAt": "2021-11-22T00:44:40.329Z",
        "__v": 0
    },
    "message": "Successfully retrieved user feedback"
}
```

**Request**
GET `/api/userFeedback/user/:id` Returns all feedback for a specific user 

**Response**
HTTP/1.1 200 OK
```
{
    "success": true,
    "message": "Feedback retrieved successfully",
    "result": [
        {
            "stars": 5,
            "reasnoning": "pog",
            "_id": "619ae7f8a78eee20210fc94e",
            "staffId": "6181f2b2bccaa891952ce94b",
            "userId": "6181f2bdfe8a2e3f44f13c18",
            "pitchId": "6181f343c6cf988c1a05b941",
            "createdAt": "2021-11-22T00:44:40.329Z",
            "updatedAt": "2021-11-22T00:44:40.329Z",
            "__v": 0
        },
        {
            "stars": 5,
            "reasnoning": "reasoning is uwu",
            "_id": "619af458d10c1121e4926dac",
            "staffId": "6181f2b2bccaa891952ce94b",
            "userId": "6181f2bdfe8a2e3f44f13c18",
            "pitchId": "6181f343c6cf988c1a05b941",
            "createdAt": "2021-11-22T01:37:28.342Z",
            "updatedAt": "2021-11-22T01:37:28.342Z",
            "__v": 0
        }
    ]
}
```

**Request**
PUT `/api/userFeedback/:id` Returns an updated user feedback
```
{
    "staffId": "6181f2b2bccaa891952ce94b",
    "userId": "6181f2bdfe8a2e3f44f13c18",
    "pitchId": "6181f343c6cf988c1a05b941",
    "stars": 4,
    "reasnoning": "pog - updated"
}
```

**Response**
HTTP/1.1 200 OK
```
{
    "success": true,
    "message": "Feedback updated successfully",
    "result": {
        "stars": 4,
        "reasnoning": "pog - updated",
        "_id": "619ae7f8a78eee20210fc94e",
        "staffId": "6181f2b2bccaa891952ce94b",
        "userId": "6181f2bdfe8a2e3f44f13c18",
        "pitchId": "6181f343c6cf988c1a05b941",
        "createdAt": "2021-11-22T00:44:40.329Z",
        "updatedAt": "2021-11-22T02:19:12.818Z",
        "__v": 0
    }
}
```

**Request**
DELETE `/api/userFeedback/:id` Deletes a user feedback

**Response**
HTTP/1.1 200 OK

```
{
    "success": true,
    "message": "Feedback deleted successfully"
}
```

### CODE REVIEW

File: api/src/api/userFeedback.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import express, { Request, Response } from 'express';
+import { errorWrap } from '../middleware';
+
+import { requireAdmin } from '../middleware/auth';
```
Comment (amit-sawhney) [inline api/src/api/userFeedback.ts:4]: Put these imports together and 1 line away from the package import.

File: api/src/models/userFeedback.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * Mongoose Schema to represent Feedback for User at South Side Weekly.
+ */
+const UserFeedback = new mongoose.Schema(
+  {
+    staffId: { type: Schema.Types.ObjectId, ref: 'User', required: true },
+    userId: { type: Schema.Types.ObjectId, ref: 'User', required: true },
+    pitchId: { type: Schema.Types.ObjectId, ref: 'Pitch', required: true },
+    stars: { type: Number, default: null },
+    reasnoning: { type: String, default: null },
```
Comment (amit-sawhney) [inline api/src/models/userFeedback.ts:15]: Typo

**Review by amit-sawhney — APPROVED** [review]
This looks really good Zora, well done integrating this with the User.

**Review by ajain1921 — APPROVED** [review]
looks good :)

**Pull request merged**